### ISSUE TITLE
Missing permissions and privileges for Pivot Role for Linking accounts

### ISSUE THREAD

**Issue by mourya-33** [awslabs/aws-dataall#528]
### Describe the bug

The dataallPivotRole cloudformation template provided on the dataall UI when linking environment is missing the following permissions and privileges to use the ML Studio and Notebooks features .

1. Trust relation for sagemaker.amazonaws.com
2. sagemaker:createApp permission.

### How to Reproduce

1. Login to the dataall UI and navigate to Organizations > Environments
2. Click on "Link Environment"
3. Bootstrap CDK using the command provided.
4. At the top of the page download the "CloudFormationTemplate" from the link and deploy it to the linked account.
5. Enter the details for  Environment -  Account ID, region and enable "Notebooks" and "ML Studio" features and click on "Create Environment".
6. Once the environment is ready, navigate to Notebooks and create a notebook. Check the stack status for the error on missing permissions.
7. Navigate to ML Studio and create a profile.
8. After the profile is created, login to the linked environment account console and navigate to Sagemaker studio, select the profile and click on OpenStudio. The studio UI will show an error for missing createApp permission on dataallPivotRole.

### Expected behavior

The stacks to create notebook and start the studio should not error out.

### Your project

dataall

### Screenshots

<img width="469" alt="Screen Shot 2023-06-20 at 3 54 24 PM" src="https://github.com/awslabs/aws-dataall/assets/134511711/0a278af8-4098-4a4c-8cad-2801df341269">
<img width="743" alt="Screen Shot 2023-06-20 at 3 54 15 PM" src="https://github.com/awslabs/aws-dataall/assets/134511711/0ebda17c-d659-421a-a519-d6fa108b3273">
<img width="1047" alt="Screen Shot 2023-06-20 at 3 54 04 PM" src="https://github.com/awslabs/aws-dataall/assets/134511711/53b84997-a909-4d95-bf6f-61c4a3964d9c">

### OS

All

### Python version

3.10

### AWS data.all version

v1.3,v1.4,v1.5

**Comment by dlpzx** [comment]
Hi @mourya-33 thanks for opening an issue!
I am trying to reproduce your scenario in a v1.5 deployment. Here is a screenshot of the pivotRole SageMaker permissions in CloudFormation. As you can see no 'sagemaker:createApp' is added:

![image](https://github.com/awslabs/aws-dataall/assets/71252798/43396184-154b-424e-82a3-3071b09c3a73)

Ans as you pointed out, sagemaker is not part of the trust policy:

![image](https://github.com/awslabs/aws-dataall/assets/71252798/b0399f89-2cab-4e71-b25e-d16806305ee8)

In an environment with Notebooks and ML Studio enabled I create a Notebook without issues:
![image](https://github.com/awslabs/aws-dataall/assets/71252798/5b98dff5-936a-4acd-9c0e-266195122161)




Then I create a SageMaker user profile, the CloudFormation stack gets created without issue
![image](https://github.com/awslabs/aws-dataall/assets/71252798/daa62ac4-ed26-4551-922b-dd3238ce44bd)

I open SageMaker Studio directly from data.all UI without issue


Then I login to the linked account from the data.all UI assuming the team Role and I go to SageMaker Studio assuming the role
Maybe the issue is here `After the profile is created, login to the linked environment account console and navigate to Sagemaker studio, select the profile and click on OpenStudio. The studio UI will show an error for missing createApp permission on dataallPivotRole.` ---> which role are you using?

**Comment by mourya-33** [comment]
Tested this again without providing the pivot role in the IAM Role Name field and it works as expected @dlpzx. As you suggested, it will be helpful if we can add additional detail on documentation or the input field itself.

Closing this issue.
